vizro-core/src/vizro/models/_components/button.py

```diff
@@ -84,16 +97,40 @@ def _action_outputs(self) -> dict[str, _IdProperty]:
     @_log_call
     def build(self):
         variants = {"plain": "link", "filled": "primary", "outlined": "secondary"}
-        description = self.description.build().children if self.description else [None]
+        description = self._build_description()
+        icon = (
+            html.Span(self.icon, id=f"{self.id}-icon", className="material-symbols-outlined tooltip-icon")
+            if self.icon
+            else None,
+        )
 
         defaults = {
             "id": self.id,
-            "children": html.Span([self.text, *description], className="button-text"),
+            "children": html.Span([*icon, self.text, *description], className="btn-text"),
             "href": get_relative_path(self.href) if self.href.startswith("/") else self.href,
             "target": "_top",
             # dbc.Button includes `btn btn-primary` as a class by default and appends any class names provided.
             # To prevent unnecessary class chaining, the button's style variant should be specified using `color`.
             "color": variants[self.variant],
+            "class_name": "btn-circular" if self.icon and not self.text else "",
```

Comment:
```suggestion
            "class_name": "btn-circular" if not self.text else "",
```

Should do the same thing I think? Given that you must provide at least one of `icon` and `text`.